jobserver/models.py

```diff
@@ -64,13 +65,28 @@ def repo_name(self):
 
 class JobQuerySet(models.QuerySet):
     def completed(self):
-        return self.filter(completed_at__isnull=False)
-
-    def in_progress(self):
-        return self.filter(started_at__isnull=False, completed_at=None)
+        return (
+            self.filter(started=True)
+            .exclude(completed_at=None)
+            .exclude(status_code=None)
+        )
+
+    def finished(self):
+        return self.exclude(started=True, completed_at=None)
+
+    def failed(self):
+        return (
+            self.exclude(completed_at=None)
+            .exclude(status_code=None)
+            .exclude(status_code__in=[6, 7, 8])
+        )
 
     def pending(self):
-        return self.filter(started_at=None, completed_at=None)
+        qwargs = Q(status_code=None) | Q(status_code__in=[6, 8])
```

Comment:
If we're using `Q` filters can we combine the `started` into the same thing, rather than mixing keyword filter argments and `Q` filters?

Reply:
I'd rather not.  We'd have to duplicated it into each side of the Q instances.